### PR TITLE
Fix & Update Pattern Library

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
1. Buttons work as expected, split buttons display correctly
2. Need to add ` onClick={this.props.onClick}` to button component
3. Add Button Group component
4. Add stepper component
5. Add Sticky Footer component
6. Update Select component
7. Update inputs & store state outside of components
8. Add Datepicker component

### CODE REVIEW

File: application/ui/components/buttons/button-group.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* jshint globalstrict: true */
+'use strict';
+
+var React       = require('react');
+var Button      = require('../buttons/button');
+var Icon        = require('../icon/icon');
```
Comment (nearengine) [inline application/ui/components/buttons/button-group.jsx]: Alignment

File: application/ui/components/buttons/button-group.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+module.exports = React.createClass({
+
+    displayName : 'ButtonGroup',
+
+    propTypes: {
```
Comment (nearengine) [inline application/ui/components/buttons/button-group.jsx]: Space after propTypes

File: application/ui/components/pattern-library/sections/pl-buttons.jsx [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 var React           = require('react');
 var Button          = require('../../buttons/button');
+var ButtonGroup     = require('../../buttons/button-group');
 var Icon            = require('../../icon/icon');
```
Comment (nearengine) [inline application/ui/components/pattern-library/sections/pl-buttons.jsx]: Alignment

File: application/ui/components/pattern-library/sections/pl-buttons.jsx [hunk 4]
```diff
@@ -125,7 +126,14 @@ module.exports = React.createClass({
                         <br />
                     </div>
                 </div>
-
+                <hr className='pl-hr' />
+                <h2 className='pl-h2'>{'Button Group'}</h2>
+                <ButtonGroup
+                    buttonOne = '#'
```
Comment (nearengine) [inline application/ui/components/pattern-library/sections/pl-buttons.jsx]: Alignment

**Comment by nearengine** [comment]
:interrobang: Marked a couple super minor alignment issues, didn't look at scss stuff

**Comment by natestormer** [comment]
:boom: see #101

**Pull request closed without merging**